modules/service/data-integrity/data-integrity-resolver.js

```diff
@@ -0,0 +1,25 @@
+const EllipticDataIntegrityService = require('./elliptic-data-integrity-service');
+
+class DataIntegrityResolver {
+    constructor() {
+        if (!DataIntegrityResolver.instance) {
+            DataIntegrityResolver.instance = this;
+        }
+
+        return DataIntegrityResolver.instance;
+    }
+
+    static getInstance() {
+        return new DataIntegrityResolver();
+    }
```

Comment:
We're instantiating a new Data integrity service every time we want to use it?

Reply:
We are using singleton pattern so it is instantiated only once